reader.c

```diff
@@ -86,20 +87,21 @@ static void nsq_reader_lookupd_poll_cb(EV_P_ struct ev_timer *w, int revents)
     ev_timer_again(rdr->loop, &rdr->lookupd_poll_timer);
 }
 
-struct NSQReader *new_nsq_reader(struct ev_loop *loop, const char *topic, const char *channel,
+struct NSQReader *new_nsq_reader(struct ev_loop *loop, const char *topic, const char *channel, void *processor,
     void (*connect_callback)(struct NSQReader *rdr, struct NSQDConnection *conn),
     void (*close_callback)(struct NSQReader *rdr, struct NSQDConnection *conn),
-    void (*msg_callback)(struct NSQReader *rdr, struct NSQDConnection *conn, struct NSQMessage *msg))
+    void (*msg_callback)(struct NSQReader *rdr, struct NSQDConnection *conn, struct NSQMessage *msg, void *processor))
```

Comment:
that can be a function pointer or c++ object pointer, in order to process the message.